unity/Assets/Scripts/ArtGallery/Util/RotateLineLighthouseToLighthouseVisibility.cs

```diff
@@ -0,0 +1,364 @@
+            {
+                // If the vertex can be seen by one other vertex add it to the 
+                // list
+                if (dic.ContainsKey(vertex2))
+                {
+                    result.Add(vertex2);
+                }
+            }
+
+            // Return the list with all vertexes that can see the vertex
+            return result;
+        }
+
+        public ICollection<Vector2> VisibleVertices(
+            Polygon2D polygon,
+            Vector2 vertex)
+        {
+            List<Vector2> result = new List<Vector2>();
+
+            // calculate new visibility polygon
+            var vision = Visibility.Vision(polygon, vertex);
+
+            if (vision == null)
+            {
+                throw new Exception("Vision polygon cannot be null");
+            }
+
+            // Check if the polygon is in clockwise order, if not, make 
+            // the polygon in clockwise order
+            if (!polygon.IsClockwise())
+            {
+                polygon.Reverse();
+            }
+
+            // Check if the vision polygon is in clockwise order, if not, make 
+            // the vision polygon in clockwise order
+            if (!vision.IsClockwise())
+            {
+                vision.Reverse();
+            }
+
+            List<Vector2> polyVertexes = polygon.Vertices.ToList();
+
+            List<Vector2> visibilityVertexes = vision.Vertices.ToList();
+
+            polyVertexes = polyVertexes.StartAt(vertex).ToList();
+            visibilityVertexes = visibilityVertexes.StartAt(vertex).ToList();
+
+            // move all vertexes such that the vertex vertex is at the origin
+            // and transform them to PolarPoint2D
+
+            var polyPolarPoints = polyVertexes
+                                  .Select(v => v - vertex)
+                                  .Select(x => new PolarPoint2D(x))
+                                  .ToList();
+
+            var visPolarPoints = visibilityVertexes
+                                 .Select(v => v - vertex)
+                                 .Select(x => new PolarPoint2D(x))
+                                 .ToList();
+
+            var initAngle = polyPolarPoints[1].Theta;
+
+            // rotate all points of the shifted polygon clockwise such that v0 lies
+            // on the x axis
+            foreach (var curr in polyPolarPoints)
+            {
+                if (!curr.IsOrigin())
+                {
+                    curr.RotateClockWise(initAngle);
+                }
+            }
+
+            foreach (var curr in visPolarPoints)
+            {
+                if (!curr.IsOrigin())
+                {
+                    curr.RotateClockWise(initAngle);
+                }
+            }
+
+            bool done = false;
+            int polyIndex = 0;
+            int visIndex = 0;
+            int polyCount = polyVertexes.Count;
+            int visCount = visibilityVertexes.Count;
+
+            while (!done)
```

Comment:
Would it be possible to use the `LineSweep` class for this algorithm so that we aren't duplicating line sweeps everywhere?

Reply:
My first instinct was to use a line sweep. But later changed this to something else. I should rename the class to depict this. Will change the name of the class.